src/x11.rs

```diff
@@ -123,14 +124,11 @@ impl GlContext {
             0,
         ];
 
+        let shared_context = shared_context
+            .map(|x| x.context)
+            .unwrap_or(std::ptr::null_mut());
         let context = unsafe {
-            glXCreateContextAttribsARB(
```

Comment:
I don't have a machine to test x11 unfortunately, so I can't guarantee this works. I have tested windows/macOS.